drivers/overlay/ostweaks_linux.go

```diff
@@ -21,10 +45,28 @@ func writeSystemProperty(key, value string) error {
 	return ioutil.WriteFile(path.Join("/proc/sys", keyPath), []byte(value), 0644)
 }
 
+func readSystemProperty(key string) ([]byte, error) {
+	keyPath := strings.Replace(key, ".", "/", -1)
+	return ioutil.ReadFile(path.Join("/proc/sys", keyPath))
+}
+
 func applyOStweaks() {
-	for k, v := range sysctlConf {
-		if err := writeSystemProperty(k, v); err != nil {
-			logrus.Errorf("error setting the kernel parameter %s = %s, err: %s", k, v, err)
+	for k, v := range osConfig {
+		// read the existing property from disk
+		oldv, err := readSystemProperty(k)
+
+		if err != nil {
+			logrus.Errorf("error reading the kernel parameter %s = %s, err: %s", k, v, err)
+			continue
+		}
+
+		if newValIsHigher(string(oldv), v.value, v.checkFn) {
+			// write new prop value to disk
+			if err := writeSystemProperty(k, v.value); err != nil {
+				logrus.Errorf("error setting the kernel parameter %s = %s, err: %s", k, v.value, err)
```

Comment:
maybe here you can also mention the old value that is maintained

Reply:
👍